### PR TITLE
coingecko,daogovernance,treasury api

### PR DESCRIPTION
**Pull request by LxzLLL**
need to be reviewed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +38,7 @@
     "prettier": "^2.5.1",
     "prettier-plugin-tailwindcss": "^0.1.6",
     "tailwindcss": "^3.0.19",
+    "tunnel": "0.0.6",
```
Comment (Darkhorse-Fraternity) [inline package.json:41]: is right in devdependce?

File: pages/api/coingecko/[id].ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      .then((response) => response.data)
+  }
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<Data>
+) {
+  const { id } = req.query
+  if (typeof id === 'string' && id == 'daosquare') {
```
Comment (Darkhorse-Fraternity) [inline pages/api/coingecko/[id].ts:38]: replace '==' to ' ==='

File: pages/api/coingecko/[id].ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    const fetchData = await fetchCoinGeckoData(id)
+    if (fetchData !== null && fetchData != undefined) {
```
Comment (Darkhorse-Fraternity) [inline pages/api/coingecko/[id].ts:40]: you can use  [!fetchData] replace

File: pages/api/daogovernance/[id].ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    },
+  })
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<Data>
+) {
+  const { id } = req.query
+  if (typeof id === 'string' && id == 'daosquare') {
```
Comment (Darkhorse-Fraternity) [inline pages/api/daogovernance/[id].ts:123]: repalce '=='

File: pages/api/daogovernance/[id].ts [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    const membersData = await fetchDaoMemberData(id)
+    const proposalsData = await fetchDaoProposalsData(id)
+
+    console.log(proposalsData)
```
Comment (Darkhorse-Fraternity) [inline pages/api/daogovernance/[id].ts:127]: remove log

File: pages/api/daogovernance/[id].ts [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+      // 已通过的提案
+      passedCount = proposalsData.data.proposals.filter(
+        (element) => element.processed == true
+      ).length
+    }
+
+    const retData: Data = {
+      id: 'daosquare',
+      members:
+        membersData != null && membersData != undefined
```
Comment (Darkhorse-Fraternity) [inline pages/api/daogovernance/[id].ts:161]: use !memberData

File: pages/api/treasury/[id].ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+const fetchTreasuryData = async (daoName: string) => {
+  return axios.get(gnosis_url).then((response) => response.data)
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<Data>
+) {
+  const { id } = req.query
+  if (typeof id === 'string' && id == 'daosquare') {
```
Comment (Darkhorse-Fraternity) [inline pages/api/treasury/[id].ts:38]: replace '=='

File: pages/api/treasury/[id].ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    const data = await fetchTreasuryData(id)
+    if (data != null && data != undefined) {
```
Comment (Darkhorse-Fraternity) [inline pages/api/treasury/[id].ts:40]: use !data

**Review by Darkhorse-Fraternity — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (LxzLLL) [reply package.json:41]: yeah

File: pages/api/coingecko/[id].ts (hunk 2 above)
Comment (LxzLLL) [reply pages/api/coingecko/[id].ts]: ok

**Pull request merged**